arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -105,6 +109,39 @@ bool findNotDoneActions(std::shared_ptr<maintenance::Action> const& action) {
 
 }  // namespace
 
+arangodb::Result arangodb::maintenance::collectionCount(arangodb::LogicalCollection const& collection,
+                                                        uint64_t& c) {
+  std::string collectionName(collection.name());
+  transaction::StandaloneContext ctx(collection.vocbase());
+  SingleCollectionTransaction trx(
+    std::shared_ptr<transaction::Context>(
+      std::shared_ptr<transaction::Context>(), &ctx),
+    collectionName, AccessMode::Type::READ);
+
+  Result res = trx.begin();
+  if (res.fail()) {
+    LOG_TOPIC("5be16", ERR, Logger::MAINTENANCE) << "Failed to start count transaction: " << res;
+    return res;
+  }
+
+  OperationOptions options(ExecContext::current());
+  OperationResult opResult =
+      trx.count(collectionName, arangodb::transaction::CountType::Normal, options);
+  res = trx.finish(opResult.result);
+
+  if (res.fail()) {
+    LOG_TOPIC("26ed2", ERR, Logger::MAINTENANCE)
+        << "Failed to finish count transaction: " << res;
+    return res;
+  }
+
+  VPackSlice s = opResult.slice();
+  TRI_ASSERT(s.isNumber());
+  c = s.getNumber<uint64_t>();
```

Comment:
```suggestion
  c = s.getNumber<int64_t>();
  if (c < 0) {
    LOG_TOPIC(...);
    return ...;
  }
```

Reply:
Just a suggestion. Not sure this would even be a problem.

Reply:
No, counts are created to be `uint64_t`, negative results are not possible.